File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/capi/ansi/keyx/rsa/Encoding.java
package aladdin.capi.ansi.keyx.rsa;

///////////////////////////////////////////////////////////////////////////
// Дополнение нулями
///////////////////////////////////////////////////////////////////////////
public class Encoding 
{
    // закодировать данные
    public static byte[] encode(byte[] data, int k)
    {
        // проверить размер данных
        if (data.length > k - 11) throw new IllegalArgumentException();

        // выделить память для результата
        byte[] encoded = new byte[k]; 

        // скопировать данные
        System.arraycopy(data, 0, encoded, k - data.length, data.length); 

        // выполнить дополнение нулями
        for (int i = 0; i < k - data.length; i++) encoded[i] = 0x00; return encoded; 
    }
}

File: Source/Aladdin.CAPI.ANSI/CAPI/ANSI/Sign/RSA/PKCS1/Encoding.cs
using System;
using System.IO;

namespace Aladdin.CAPI.ANSI.Sign.RSA.PKCS1
{
    ///////////////////////////////////////////////////////////////////////////
    // Кодирование EMSA PKCS1
    ///////////////////////////////////////////////////////////////////////////
    public static class Encoding
    {
        // закодировать данные
	    public static byte[] Encode(byte[] data, int emLen)
	    {
		    // проверить размер данных
		    if (data.Length > emLen - 11) throw new InvalidDataException();

		    // выделить память для результата
		    byte[] encoded = new byte[emLen]; encoded[0] = 0x00; encoded[1] = 0x01;

		    // установить специальное заполнение
		    for (int i = 2; i < emLen - data.Length - 1; i++) encoded[i] = 0xFF; 

		    // скопировать данные
		    Array.Copy(data, 0, encoded, emLen - data.Length, data.Length); 
			
		    // записать значение разделителя
		    encoded[emLen - data.Length - 1] = 0x00; return encoded; 
	    }
    }
}
